rero_ils/modules/loans/extensions.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""RERO ILS common record extensions."""
+
+
+from datetime import timedelta, timezone
+
+import ciso8601
+from invenio_records.extensions import RecordExtension
+
+
+class SetDueSoonDate(RecordExtension):
+    """Set the due soon date when the loan is updated."""
+
+    def pre_commit(self, record):
```

Comment:
From my memory : the `pre_commit` hook is only used when you **update** a record. 
If the `due_soon_date` should be created anytime, you also need to use the **pre_create** hook.
Something like 
```python

def _create_due_soon(self, record):
    ...
    
pre_commit = _create_due_soon
pre_create = _create_due_soon
```

Reply:
I do not think so. A loan can becomes ON_LOAN only at update, not during the creation.

Reply:
You're right. I hope it will be always the case :-)